### PR TITLE
AP_TemperatureSensor: fix param count change when loading drivers

### PR DESCRIPTION
**Pull request by magicrub**
fix AP_TemperatureSensor param count change when loading drivers which was causing GCS spam. Closes  https://github.com/ArduPilot/ardupilot/issues/22357

### CODE REVIEW

**Comment by magicrub** [comment]
poke @andyp1per or @rmackay9 or @Williangalvani . This should be an easy merge for an annoying gcs spam bug for sub

File: libraries/AP_TemperatureSensor/AP_TemperatureSensor.cpp [hunk 1]
```diff
@@ -142,6 +142,9 @@ void AP_TemperatureSensor::init()
 
         // call init function for each backend
         if (drivers[instance] != nullptr) {
+            // param count could have changed
+            AP_Param::invalidate_count();
```
Comment (andyp1per) [inline libraries/AP_TemperatureSensor/AP_TemperatureSensor.cpp:146]: Why not call this at the end rather than inside the loop?

Comment (magicrub) [reply libraries/AP_TemperatureSensor/AP_TemperatureSensor.cpp:146]: I could..

Comment (magicrub) [reply libraries/AP_TemperatureSensor/AP_TemperatureSensor.cpp]: changed to only run once

**Review by Williangalvani — APPROVED** [review]

**Pull request merged**